include/simdjson/inline/parser.h

```diff
@@ -101,19 +93,14 @@ inline simdjson_result<document_stream> parser::load_many(const std::string &pat
 inline simdjson_result<element> parser::parse(const uint8_t *buf, size_t len, bool realloc_if_needed) & noexcept {
   error_code _error = ensure_capacity(len);
   if (_error) { return _error; }
+  std::unique_ptr<uint8_t[]> tmp_buf;
 
   if (realloc_if_needed) {
-    const uint8_t *tmp_buf = buf;
-    buf = (uint8_t *)internal::allocate_padded_buffer(len);
-    if (buf == nullptr)
-      return MEMALLOC;
-    memcpy((void *)buf, tmp_buf, len);
-  }
-
-  _error = implementation->parse(buf, len, doc);
-  if (realloc_if_needed) {
-    aligned_free((void *)buf); // must free before we exit
+    tmp_buf.reset((uint8_t *)internal::allocate_padded_buffer(len));
```

Comment:
With the change `allocate_padded_buffer` requires delete[] to be used to free the memory, eg `std::unique_ptr<uint8_t[]>` is ok here. The problem might happen if  `allocate_padded_buffer` is updated later to use malloc for example.

Reply:
Of course.